src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -0,0 +1,133 @@
+package plugins;
+
+import jakarta.inject.Inject;
+import org.gitlab4j.api.GitLabApiException;
+import org.jenkinsci.test.acceptance.docker.DockerContainerHolder;
+import org.jenkinsci.test.acceptance.docker.fixtures.GitLabContainer;
+import org.jenkinsci.test.acceptance.junit.*;
+import org.jenkinsci.test.acceptance.plugins.credentials.CredentialsPage;
+import org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials;
+import org.jenkinsci.test.acceptance.plugins.gitlab_plugin.*;
+import org.jenkinsci.test.acceptance.po.WorkflowMultiBranchJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.net.http.HttpResponse;
+
+import java.io.IOException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.jenkinsci.test.acceptance.Matchers.hasContent;
+import static org.junit.Assert.*;
+
+@WithDocker
+@Category(DockerTest.class)
+@WithPlugins({"gitlab-branch-source", "workflow-multibranch"})
+public class GitLabPluginTest extends AbstractJUnitTest {
+
+    @Inject
+    DockerContainerHolder<GitLabContainer> gitLabServer;
+
+    private GitLabContainer container;
+    private String repoUrl;
+    private String host;
+    private int port;
+
+    private String privateToken;
+
+    private String password = "arandompassword12#"; // I guess the password can be the same for all users
```

Comment:
ideally you would want it to be different to ensure that things do work as expected.